python/cudf/cudf/_lib/pylibcudf/aggregation.pyx

```diff
@@ -85,40 +83,43 @@ cdef class Aggregation:
         """Get the kind of the aggregation."""
         return dereference(self.c_obj).kind
 
-    cdef unique_ptr[groupby_aggregation] clone_underlying_as_groupby(self) except *:
-        """Make a copy of the underlying aggregation that can be used in a groupby.
+    cdef agg_ptr _raise_if_null(self, agg_ptr ptr, str alg):
+        # The functions calling this all use a dynamic cast between aggregation types,
+        # and the cast returning a null pointer is how we capture whether or not
+        # libcudf supports a given aggregation for a particular algorithm.
+        if ptr is NULL:
+            agg_repr = str(self.kind()).split(".")[1].title()
+            raise TypeError(f"{agg_repr} aggregations are not supported by {alg}")
```

Comment:
I'd maybe reduce the scope of this helper:

```cython
cdef _unsupported_agg_error(self, alg):
    agg_repr = str(self.kind()).split(".")[1].title()
    raise TypeError(f"{agg_repr} aggregations are not supported by {alg}")   
```

Then:

```cython
cdef groupby_scan_aggregation *agg_cast = dynamic_cast[gbsa_ptr](agg.get())
if agg_cast is NULL:
    self._unsupported_agg_error("groupby scan")
```

This way we don't have to define the fused type `agg_ptr`

Reply:
It's kinda subjective. Let me know if you think differently..

Reply:
Ha I actually had that originally and changed it to this to make the error consistent and reduce duplication, but then it adds the need for the fused type, which I agree feels heavy-handed here. Should we maybe just get rid of the helper entirely and just inline the error check?

Reply:
Discussed offline. Going with Ashwin's suggestion.